src/metabase/models/card.clj

```diff
@@ -237,6 +239,49 @@
       (field-values/update-field-values-for-on-demand-dbs! field-ids))
     (create-actions-when-is-writable! card)))
 
+(defn- assoc-viz-setting
+  [card key val]
+  (assoc-in card [:visualization_settings key] val))
+
+(defmulti ^:private migrate-viz-settings*
+  "Migration visualization_settings to the appropriate version"
+  {:arglists '([card desired-version])}
+  (fn [card desired-version]
+    (let [current-version (or (get-in card [:visualization_settings :version])
+                              1)]
+      (if (= current-version desired-version)
+        ::identity
+        [current-version desired-version]))))
+
+(defmethod ^:private migrate-viz-settings* ::identity [card _]
+  card)
+
+(defmethod ^:private migrate-viz-settings* [1 2] [{:keys [visualization_settings] :as card} _]
+  (let [{percent? :pie.show_legend_perecent ;; [sic]
+         legend?  :pie.show_legend} visualization_settings]
+    (assoc-viz-setting card :pie.percent_visibility
+                       (cond
+                         legend?  "inside"
+                         percent? "legend"
+                         :else    "off"))))
+
+;; migrate-viz settings was introduced with v. 2, so we'll never be in a situation where we can downgrade from 2 to 1.
```

Comment:
I think you mentioned somewhere in a Slack thread that you could backport the migrate-viz-settings function(s)/methods right? 

If you were to go down that road, then it would be possible to create the `[2 1]` migration right? Where V1. is actually any viz-settings map that does NOT contain a :version key?

Just opening this for discussion and understanding on my side, so if it's silly, disregard of course.

Reply:
> If you were to go down that road, then it would be possible to create the [2 1] migration right? Where V1. is actually any viz-settings map that does NOT contain a :version key?

Yes. I'm still torn on whether it's worth the effort to do the backport stuff. Any opinions welcome

Reply:
I think probably not, actually. I think if there's a downgrade situation, the frontend should save appropriate viz-settings back into the card when its viewed next.